git_machete/cmd.py

```diff
@@ -339,362 +339,1207 @@ def popen_git(cli_ctxt: CommandLineContext, git_cmd: str, *args: str, **kwargs:
+        # Even worse, reset's reflog message would be filtered out in our fork point algorithm,
+        # so the squashed commit would not even be considered to "belong"
+        # (in the FP sense) to the current branch's history.
+        run_git(self.cli_ctxt, "update-ref", "HEAD", squashed_sha, "-m", f"squash: {earliest_subject}")
+
+        print(f"Squashed {len(commits)} commits:")
+        print()
+        for sha, short_sha, subject in commits:
+            print(f"\t{short_sha} {subject}")
+
+        latest_sha, latest_short_sha, latest_subject = commits[-1]
+        print()
+        print("To restore the original pre-squash commit, run:")
+        print()
+        print(fmt(f"\t`git reset {latest_sha}`"))
+
+    def filtered_reflog(self, b: str, prefix: str) -> List[str]:  # add
+        def is_excluded_reflog_subject(sha_: str, gs_: str) -> bool:
+            is_excluded = (
+                    gs_.startswith("branch: Created from") or
+                    gs_ == f"branch: Reset to {b}" or
+                    gs_ == "branch: Reset to HEAD" or
+                    gs_.startswith("reset: moving to ") or
+                    gs_.startswith("fetch . ") or
+                    # The rare case of a no-op rebase, the exact wording likely depends on git version
+                    gs_ == f"rebase finished: {prefix}{b} onto {sha_}" or
+                    gs_ == f"rebase -i (finish): {prefix}{b} onto {sha_}"
+            )
+            if is_excluded:
+                debug(self.cli_ctxt, f"filtered_reflog({b}, {prefix}) -> is_excluded_reflog_subject({sha_}, <<<{gs_}>>>)",
+                      "skipping reflog entry")
+            return is_excluded
+
+        b_reflog = reflog(self.cli_ctxt, prefix + b)
+        if not b_reflog:
+            return []
+
+        earliest_sha, earliest_gs = b_reflog[-1]  # Note that the reflog is returned from latest to earliest entries.
+        shas_to_exclude = set()
+        if earliest_gs.startswith("branch: Created from"):
+            debug(self.cli_ctxt,
+                  f"filtered_reflog({b}, {prefix})",
+                  f"skipping any reflog entry with the hash equal to the hash of the earliest (branch creation) entry: {earliest_sha}")
+            shas_to_exclude.add(earliest_sha)
+
+        result = [sha for (sha, gs) in b_reflog if
+                  sha not in shas_to_exclude and not is_excluded_reflog_subject(sha, gs)]
+        debug(self.cli_ctxt,
+              f"filtered_reflog({b}, {prefix})",
+              "computed filtered reflog (= reflog without branch creation "
+              "and branch reset events irrelevant for fork point/upstream inference): %s\n" % (
+                          ", ".join(result) or "<empty>"))
+        return result
 
 
-def sync_annotations_to_github_prs(cli_ctxt: CommandLineContext) -> None:
-    global annotations, managed_branches
+def sync_annotations_to_github_prs(machete_client: MacheteClient) -> None:  # ask if member of MacheteClient
```

Comment:
:yes:

Reply:
I changed my mind on this one, as the general idea of `MacheteContext` changed more into `MacheteClient` 😅